react-native-gutenberg-bridge/android/build.gradle

```diff
@@ -181,6 +181,38 @@ if (buildGutenbergMobileJSBundle) {
     def bundleName = 'index.android.bundle'
     def jsRootDir = '../..'
 
+    // FIXME also check that submodule is properly up to date???
+    task abortIfGutenbergMobileSubmoduleNotClean(type: Exec) {
+        workingDir "../../"
+        commandLine "git", "status", "--porcelain"
+        standardOutput = new ByteArrayOutputStream()
+
+        // Storing this because we cannot access `standardOutput` in the doLast block
+        def output = { standardOutput.toString() }
+        doLast {
+            if (!output().isEmpty()) {
+                def message = "Attempting to build Gutenberg JS Bundle without a clean submodule. \
+Either clean the `libs/gutenberg-mobile` submodule (you probably want to do this) or \
+build Gutenberg from source by setting wp.BUILD_GUTENBERG_FROM_SOURCE to true."
```

Comment:
WDYT about adding where that flag resides? It might not be easy to know for someone not experienced with developing in gutenberg-mobile.